helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -77,8 +77,15 @@ receivers:
 
   kubeletstats:
     collection_interval: 10s
+    {{- if eq .Values.distribution "gke/autopilot" }}
+    # GKE Autopilot doesn't allow using the secure kubelet endpoint,
+    # use the read-only endpoint instead.
+    auth_type: none
+    endpoint: ${K8S_NODE_IP}:10255
```

Comment:
Is readonly port usage documented for autopilot? I wonder how dependable usage of this is or if we're entitled to any kind of deprecation notice.

Reply:
This is kubernetes problem in general, not Autopilot. Kubelet endpoins are not documented. The secure endpoint doesn't provide a way to authorize just GET requests, that's why Autopilot blocks the whole endpoint. There are plans in k8s ecosystem to deprecate the read-only endpoint and provide a granular access to secure endpoint, once it happens we can switch to it.